Update typesetting challenge (#41384)
mdn/content: files/en-us/learn_web_development/core/text_styling/typesetting_a_homepage/index.md

{{PreviousMenuNext("Learn_web_development/Core/Text_styling/Web_fonts", "Learn_web_development/Core/CSS_layout", "Learn_web_development/Core/Text_styling")}}

> In this challenge, we'll test your understanding of the text styling techniques we've covered throughout this module by getting you to typeset a community school's homepage. You might just have some fun along the way.

Reviewer: Feels like a very long sentence. Mabye 
```suggestion
In this challenge, we'll test your understanding of the text styling techniques we've covered throughout this module — by getting you to typeset a community school's homepage. You might just have some fun along the way.
```

> You might just have some fun along the way.

If you're some kind of sicko :-)

             <li><a href="#">Staff</a></li>
             <li><a href="#">Media</a></li>
             <li><a href="#">Prospectus</a></li>
           </ul>
         </nav>
       </main>

       <footer>
         <p>&copy; 2025 St Huxley's Community College</p>
       </footer>
     </body>
   </html>
   ```

3. Inside the folder, create a `style.css` file and paste the following content into it:

   ```css
   /* General setup */

   * {
     box-sizing: border-box;
   }

   body {
     margin: 0 auto;
     padding: 0 20px;
     min-width: 980px;
     max-width: 1400px;
   }

   /* Layout */

   main {
     display: grid;
     grid-template-columns: 5fr 2fr 2fr;
     gap: 40px;
     padding: 20px 0;
   }

   /* header and footer */

   header {
     border-bottom: 5px solid #a66;
   }

   footer {
     border-top: 5px solid #a66;
   }

   footer p {
     text-align: center;
   }
   ```

4. Download the [`external-link-52.png`](https://mdn.github.io/shared-assets/images/examples/external-link-52.png) icon and save it in the folder at the same level as the code files.

5. Save your files and load `index.html` in a browser ready to test.

## Project brief

> You've been provided with some HTML for the homepage of an imaginary community college, plus some CSS that lays out the content into three columns and provides some other rudimentary styling. You need to add rules to the bottom of the CSS file to solve the challenges described in the below sections.

Reviewer: To my taste

```suggestion
You've been provided with some HTML for the homepage of an imaginary community college, plus some CSS that lays out the content into three columns and provides some other rudimentary styling. You need to add rules to the bottom of the CSS file to solve the challenges described in the following sections.
```

### Apply some fonts to the page

> 1. Choose a couple of fonts to apply to the page. Because this is a college, the fonts should give the site a fairly serious, trustworthy feel. A serif site-wide font for the general body text coupled with a heavy/slab font for the headings would work.
> 2. It is your choice whether you want to use an online font service like Google Fonts to access the fonts, or download font files locally to your project. Whatever you choose, make your fonts available to your page. If you choose to use local font files, use a suitable service to generate bulletproof `@font-face` code for them.

Reviewer: I would do it like this, because the points 1 and 2 are both to do with font selection

```suggestion
1. Choose heading and body fonts to apply to the page:
   - Because this is a college, the fonts should give the site a fairly serious, trustworthy feel. A serif site-wide font for the general body text coupled with a heavy/slab font for the headings would work.
   - It is your choice whether you want to use an online font service like Google Fonts to access the fonts, or download font files locally to your project. Whatever you choose, make your fonts available to your page. If you choose to use local font files, use a suitable service to generate bulletproof `@font-face` code for them.
```

### Lists

1. Make sure the spacing of your lists and list items works well with the styling of the overall page. Each list should have the same `line-height` and top and bottom margin as the paragraphs.
2. Give your list items appropriate bullet styles for the design of the page. It is up to you whether you choose a custom bullet image or something else.

### Navigation menu

Style your navigation menu so that it harmonizes with the page. We'll leave this largely up to you, but here are some tips/suggestions:

1. Make the links look like buttons, which are as wide as the column they are in, and tall enough that the nav items fill a decent amount of space.
2. Apply the same font to your nav link text as you applied to your headings.
3. Make sure the hit area of each link is expanded to fill the entirety of its parent list item.
4. Center the text inside each link.
5. Uppercase the text (using CSS, not by editing the HTML!)

## Hints and tips

- You don't need to edit the HTML in any way for this exercise, unless required to apply the fonts to the page.

## Example

The following screenshot shows an example of what the finished design could look like:

> ![A screenshot of the finished challenge design. The top heading reads 'St Huxley's Community College'. There is a red line separating the banner header from the content. The main content has three columns, two containing text, and a vertical navigation bar in the third column.](example.png)

Author: That's a nice idea. Done!
